### PR TITLE
Migrations guide for Localized CupertinoTabBar Semantics

### PR DESCRIPTION
**Pull request by Piinks**
This a migration guide for https://github.com/flutter/flutter/pull/55336

### CODE REVIEW

File: src/docs/release/breaking-changes/cupertino-tab-bar-localizations.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+To support localized semantics information, the `CupertinoTabBar` requires localizations. (Before 
+this change, the `Semantics` hint provided to the `CupertinoTabBar` was a hard-coded String,
+'tab, $index of $total'.)
+
+If your `CupertinoTabBar` is within the scope of a `CupertinoApp`, the
+`DefaultCupertinoLocalizations` will already be instantiated and may suit your needs without having
+to make a change to your existing code.
+
+If your `CupertinoTabBar` is not within a `CupertinoApp`, you may provide the `Localizations` of
```
Comment (Piinks) [inline src/docs/release/breaking-changes/cupertino-tab-bar-localizations.md:30]:
```suggestion
If your `CupertinoTabBar` is not within a `CupertinoApp`, you may provide the localizations of
```

File: src/docs/release/breaking-changes/cupertino-tab-bar-localizations.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+your choosing using the `Localizations` widget.
+
+## Migration guide
+
+If you are seeing `'localizations != null'` assertion error, make sure locale information is being
```
Comment (Piinks) [inline src/docs/release/breaking-changes/cupertino-tab-bar-localizations.md:35]:
```suggestion
If you are seeing a `'localizations != null'` assertion error, make sure locale information is being
```

File: src/docs/release/breaking-changes/cupertino-tab-bar-localizations.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+* [`DefaultCupertinoLocalizations`][]
+* [`Semantics`][]
+* [`CupertinoApp`][]
+* [Internationalizing Flutter Apps][]
+
+
+Relevant PR:
+* [PR 55336: Adding tabSemanticsLabel to CupertinoLocalizations][]
+
+
```
Comment (Piinks) [inline src/docs/release/breaking-changes/cupertino-tab-bar-localizations.md:152]:
```suggestion
```

File: src/docs/release/breaking-changes/index.md [hunk 4]
```diff
@@ -35,6 +36,7 @@ The following guides (in alphabetical order) are available:
 [Annotations return local position relative to object]: /docs/release/breaking-changes/annotations-return-local-position-relative-to-object
 [breaking change policy]: /docs/resources/compatibility
 [Container color optimization]: /docs/release/breaking-changes/container-color
+[CupertinoTabBar requires Localizations parent]: /docs/release/breaking-changes/cupertino-tab-bar-localizations.md
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/index.md:39]: Delete the ".md" from the end. The generated file is not markdown, it's html.  Thx!

**Comment by sfshaza2** [comment]
Actually, I rescind my LGTM. I was too quick.  :D

There are two issues:`
- Delete the ".md" extension from the URL in the index file. It's the reason that Travis broke.
- We can't land these breaking change docs until we know in which build they have landed. The PR is still open, so this can't be merged.

File: src/docs/release/breaking-changes/index.md (hunk 4 above)
Comment (Piinks) [inline src/docs/release/breaking-changes/index.md:39]:
```suggestion
[CupertinoTabBar requires Localizations parent]: /docs/release/breaking-changes/cupertino-tab-bar-localizations
```

**Comment by Piinks** [comment]
This change has landed, just waiting for the build tag. :)

**Comment by sfshaza2** [comment]
I see that this landed in 1.18.0. (The actual version is  1.18.0-10.0.pre.) Can we finish up this PR, @Piinks?

File: src/docs/release/breaking-changes/cupertino-tab-bar-localizations.md [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+        ),
+      ),
+    );
+  }
+}
+```
+
+## Timeline
+
+TBD
```
Comment (Piinks) [inline src/docs/release/breaking-changes/cupertino-tab-bar-localizations.md:136]:
```suggestion
This change was introduced in May 2020, in 1.18.0-9.0.pre.
```

**Comment by Piinks** [comment]
> I see that this landed in 1.18.0. (The actual version is 1.18.0-10.0.pre.) Can we finish up this PR, @Piinks?

All set! :)

**Comment by Piinks** [comment]
@sfshaza2 it looks like this is failing on Travis from an unrelated change?

**Comment by Piinks** [comment]
I need to add mention of the hint changing, per https://github.com/flutter/flutter/pull/56582

**Comment by Piinks** [comment]
This was not ready to be merged. I'll open a new PR to update.

**Comment by Piinks** [comment]
whoops, wrong pr! :)

**Pull request merged**